optuna/integration/xgboost.py

```diff
@@ -43,20 +43,30 @@ class XGBoostPruningCallback(object):
             ``validation-merror``. Please refer to ``eval_metric`` in
             `XGBoost reference <https://xgboost.readthedocs.io/en/latest/parameter.html>`_
             for further details.
+        interval:
+            Check if trial should be pruned every n-th epoch. By default `interval=1` and
+            pruning is performed after every epoch. Increase `interval` to run several
+            epochs faster before applying pruning.
     """
 
-    def __init__(self, trial, observation_key):
-        # type: (optuna.trial.Trial, str) -> None
+    def __init__(self, trial, observation_key, interval=1):
+        # type: (optuna.trial.Trial, str, int) -> None
 
         _check_xgboost_availability()
 
         self._trial = trial
         self._observation_key = observation_key
+        self._interval = interval
 
     def __call__(self, env):
         # type: (xgb.core.CallbackEnv) -> None
 
         context = _get_callback_context(env)
+
+        # Iterations start with 1.
```

Comment:
Are you sure this always holds? It seems to start from 0 occasionally, running existing tests. I'm not sure what controls it, though.

Reply:
Actually iterations start from 0, in one of existing tests iterations start from 1 because `env.iteration` is artificially set to 1 ([link](https://github.com/optuna/optuna/blob/3fe0ab712096eaf434e521b1b0b4f0fab4dc8740/tests/integration_tests/test_xgboost.py#L20)). Comment `# Iterations start with 1.` was wrong.